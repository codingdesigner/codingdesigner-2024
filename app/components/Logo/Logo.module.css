.logo path {
  fill: var(--main-text-color);
}

.logo path[fill="#FF0000"],
.logo g[fill="#FF0000"] path {
  fill: light-dark(#e7131a, #ff7878);
}

.logo path[fill="#C6A348"],
.logo g[fill="#C6A348"] path {
  fill: #c6a348;
}

/* only Safari needs this */
@supports not (color: light-dark(red, pink)) {
  .logo path[fill="#FF0000"],
  .logo g[fill="#FF0000"] path {
    fill: #e7131a;
  }
  @media (prefers-color-scheme: dark) {
    .logo path[fill="#FF0000"],
    .logo g[fill="#FF0000"] path {
      fill: #ff7878;
    }
  }
}
