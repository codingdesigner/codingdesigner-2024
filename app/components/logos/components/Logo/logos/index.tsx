import Allure from './allure';
import ArchitecturalDigest from './architectural-digest';
import BonAppetit from './bon-appetit';
import Clever from './clever';
import Epicurious from './epicurious';
import Glamour from './glamour';
import GQ from './gq';
import Hive from './hive';
import HouseGarden from './house-garden';
import LaCucinaItalia from './la-cucina-italiana';
import NewYorker from './new-yorker';
import Pitchfork from './pitchfork';
import Self from './self';
import Tatler from './tatler';
import TeenVogue from './teen-vogue';
import Them from './them';
import Traveler from './traveler';
import VanityFaire from './vanity-faire';
import Vogue from './vogue';
import Wired from './wired';

export {
  Allure,
  ArchitecturalDigest,
  BonAppetit,
  Clever,
  Epicurious,
  Glamour,
  GQ,
  Hive,
  HouseGarden,
  LaCucinaItalia,
  NewYorker,
  Pitchfork,
  Self,
  Tatler,
  TeenVogue,
  Them,
  Traveler,
  VanityFaire,
  Vogue,
  Wired,
}
