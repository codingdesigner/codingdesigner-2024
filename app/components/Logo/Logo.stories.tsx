import React from 'react';
import type {Meta, StoryObj} from '@storybook/react';

import { Logo } from './Logo';

const meta: Meta<typeof Logo> = {
  title: 'Logos',
  component: Logo,
};

export default meta;

type Story = StoryObj<typeof Logo>;

export const Allure: Story = {args: {logoName: 'allure'}};
export const ArchitecturalDigest: Story = {args: {logoName: 'architectural-digest'}};
export const BonAppetit: Story = {args: {logoName: 'bon-appetit'}};
export const Clever: Story = {args: {logoName: 'clever'}};
export const Epicurious: Story = {args: {logoName: 'epicurious'}};
export const Glamour: Story = {args: {logoName: 'glamour'}};
export const GQ: Story = {args: {logoName: 'gq'}};
export const Hive: Story = {args: {logoName: 'hive'}};
export const HouseGarden: Story = {args: {logoName: 'house-garden'}};
export const LaCucinaItalia: Story = {args: {logoName: 'la-cucina-italiana'}};
export const NewYorker: Story = {args: {logoName: 'new-yorker'}};
export const Pitchfork: Story = {args: {logoName: 'pitchfork'}};
export const Self: Story = {args: {logoName: 'self'}};
export const Tatler: Story = {args: {logoName: 'tatler'}};
export const Them: Story = {args: {logoName: 'them'}};
export const Traveler: Story = {args: {logoName: 'traveler'}};
export const VanityFaire: Story = {args: {logoName: 'vanity-faire'}};
export const Vogue: Story = {args: {logoName: 'vogue'}};
export const Wired: Story = {args: {logoName: 'wired'}};
