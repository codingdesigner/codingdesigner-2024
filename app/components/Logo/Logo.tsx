import React from 'react';
import * as logoSvgs from './logos/index'; // Import all logos from the logos directory

// import styles from './Logo.module.css';

// Define a map of logo components, where the key is the logo name and the value is the corresponding React component
const logoComponents: { [key: string]: React.ComponentType } = {
  'allure': logoSvgs.Allure,
  'architectural-digest': logoSvgs.ArchitecturalDigest,
  'bon-appetit': logoSvgs.BonAppetit,
  'clever': logoSvgs.Clever,
  'epicurious': logoSvgs.Epicurious,
  'glamour': logoSvgs.Glamour,
  'gq': logoSvgs.GQ,
  'hive': logoSvgs.Hive,
  'house-garden': logoSvgs.HouseGarden,
  'la-cucina-italiana': logoSvgs.LaCucinaItalia,
  'new-yorker': logoSvgs.NewYorker,
  'pitchfork': logoSvgs.Pitchfork,
  'self': logoSvgs.Self,
  'tatler': logoSvgs.Tatler,
  'teen-vogue': logoSvgs.TeenVogue,
  'them': logoSvgs.Them,
  'traveler': logoSvgs.Traveler,
  'vanity-faire': logoSvgs.VanityFaire,
  'vogue': logoSvgs.Vogue,
  'wired': logoSvgs.Wired
};

// Define the props for the Logo component
export interface LogoProps {
  logoName?: string; // The name of the logo to be rendered
}

// Define the Logo component as a functional component
export const Logo: React.FC<LogoProps> = ({ logoName }) => {
  let LogoSvg
  if (logoName) {
    LogoSvg = logoComponents[logoName]; // Get the component from the map using the provided logoName
  }
  if (LogoSvg) {
    return <LogoSvg data-testid="logo"/>; // If the corresponding logo component is found, render it
  } else {
    return null; // If the logo component is not found, render nothing
  }
}
