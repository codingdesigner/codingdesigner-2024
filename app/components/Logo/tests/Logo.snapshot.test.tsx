import '@testing-library/jest-dom'
import {render} from '@testing-library/react'

import {Logo} from '../Logo';

describe('<Logo />', () => {
  // let component: ReturnType<typeof render>;

  // beforeEach(() => {
  // });

  it('renders a logo', () => {
    // const columnSet: HTMLElement = component.getByRole('image')
    const component = render(<Logo logoName={'vogue'}/>);
    expect(component).toMatchSnapshot()
  });

  it('does not render if a logo isn\'t provided', () => {
    const component = render(<Logo />);
    expect(component).toMatchSnapshot()
  });

});
