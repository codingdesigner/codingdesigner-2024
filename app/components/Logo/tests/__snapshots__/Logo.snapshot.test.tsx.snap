// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> does not render if a logo isn't provided 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Logo /> renders a logo 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class="logo"
        data-testid="logo"
        fill="none"
        height="36"
        viewBox="0 0 137 36"
        width="137"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#vogueclip)"
        >
          <path
            clip-rule="evenodd"
            d="M82.66 21.108h2.536v-.335H74.738v.335h2.692v7.389c0 3.415-1.469 6.54-6.142 6.629-3.36.044-5.184-1.853-6.163-4.487-1.069-2.812-1.447-5.78-1.447-12.5 0-4.308.4-10.692 1.87-13.549.69-1.34 1.958-3.705 5.674-3.705 4.673.022 7.387 3.84 9.39 11.83h.111V.483h-.11c-.98 2.366-4.674 1.205-4.674 1.205-.44-.139-.827-.27-1.181-.39-1.291-.437-2.144-.726-3.559-.726-5.763 0-13.685 5.87-13.685 17.991 0 9.889 5.92 16.898 14.02 16.898 1.745 0 3.131-.663 4.462-1.3 1.23-.589 2.414-1.156 3.793-1.156.89 0 2.559.558 2.826 2.01h.067l-.023-13.907ZM109.05 1.8h6.008v32.678h-2.937v.358h24.032V22.559h-.155c-.553 12.23-14.061 11.993-15.287 11.946V17.983l.441-.002c2.835-.02 6.398-.046 6.968 7.078h.112V10.66h-.134c-.757 6.92-4.584 6.965-7.41 6.965V1.866c.43 0 .862-.006 1.296-.012 5.183-.078 10.524-.159 12.701 10.861h.133V1.465H105.624V1.8h3.026v24.42c-.111 6.138-4.539 8.928-7.966 8.817-6.275-.224-6.988-6.183-6.988-7.411V1.8h3.005v-.335H85.018V1.8h3.093v22.835c0 6.116 3.093 10.848 12.15 10.848 6.075 0 8.856-5.067 8.789-9.263V1.8Zm-86.137-.023v-.312h6.119v.312h-2.648l-9.902 33.237h-.178L3.843 1.777H.816v-.312h11.06v.312H9.228l9.167 25.514 7.61-25.514h-3.092Zm25.411 16.184c0 2.366-.178 9.91-2.158 13.66-1.224 2.411-3.36 3.416-5.852 3.416-2.493 0-4.629-1.027-5.853-3.415-1.958-3.75-2.158-11.295-2.158-13.661 0-2.366.178-9.665 2.158-13.572 1.18-2.41 3.36-3.415 5.853-3.415 2.492 0 4.672 1.027 5.852 3.415 1.98 3.907 2.158 11.228 2.158 13.572Zm6.32 0c0-10.469-7.499-17.322-14.33-17.322-6.832 0-14.33 6.853-14.33 17.322s7.743 17.41 14.33 17.41c6.608 0 14.33-6.942 14.33-17.41Z"
            fill="#000"
            fill-rule="evenodd"
          />
        </g>
        <defs>
          <clippath
            id="vogueclip"
          >
            <path
              d="M.816.483h135.377v35H.816z"
              fill="#fff"
            />
          </clippath>
        </defs>
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="logo"
      data-testid="logo"
      fill="none"
      height="36"
      viewBox="0 0 137 36"
      width="137"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#vogueclip)"
      >
        <path
          clip-rule="evenodd"
          d="M82.66 21.108h2.536v-.335H74.738v.335h2.692v7.389c0 3.415-1.469 6.54-6.142 6.629-3.36.044-5.184-1.853-6.163-4.487-1.069-2.812-1.447-5.78-1.447-12.5 0-4.308.4-10.692 1.87-13.549.69-1.34 1.958-3.705 5.674-3.705 4.673.022 7.387 3.84 9.39 11.83h.111V.483h-.11c-.98 2.366-4.674 1.205-4.674 1.205-.44-.139-.827-.27-1.181-.39-1.291-.437-2.144-.726-3.559-.726-5.763 0-13.685 5.87-13.685 17.991 0 9.889 5.92 16.898 14.02 16.898 1.745 0 3.131-.663 4.462-1.3 1.23-.589 2.414-1.156 3.793-1.156.89 0 2.559.558 2.826 2.01h.067l-.023-13.907ZM109.05 1.8h6.008v32.678h-2.937v.358h24.032V22.559h-.155c-.553 12.23-14.061 11.993-15.287 11.946V17.983l.441-.002c2.835-.02 6.398-.046 6.968 7.078h.112V10.66h-.134c-.757 6.92-4.584 6.965-7.41 6.965V1.866c.43 0 .862-.006 1.296-.012 5.183-.078 10.524-.159 12.701 10.861h.133V1.465H105.624V1.8h3.026v24.42c-.111 6.138-4.539 8.928-7.966 8.817-6.275-.224-6.988-6.183-6.988-7.411V1.8h3.005v-.335H85.018V1.8h3.093v22.835c0 6.116 3.093 10.848 12.15 10.848 6.075 0 8.856-5.067 8.789-9.263V1.8Zm-86.137-.023v-.312h6.119v.312h-2.648l-9.902 33.237h-.178L3.843 1.777H.816v-.312h11.06v.312H9.228l9.167 25.514 7.61-25.514h-3.092Zm25.411 16.184c0 2.366-.178 9.91-2.158 13.66-1.224 2.411-3.36 3.416-5.852 3.416-2.493 0-4.629-1.027-5.853-3.415-1.958-3.75-2.158-11.295-2.158-13.661 0-2.366.178-9.665 2.158-13.572 1.18-2.41 3.36-3.415 5.853-3.415 2.492 0 4.672 1.027 5.852 3.415 1.98 3.907 2.158 11.228 2.158 13.572Zm6.32 0c0-10.469-7.499-17.322-14.33-17.322-6.832 0-14.33 6.853-14.33 17.322s7.743 17.41 14.33 17.41c6.608 0 14.33-6.942 14.33-17.41Z"
          fill="#000"
          fill-rule="evenodd"
        />
      </g>
      <defs>
        <clippath
          id="vogueclip"
        >
          <path
            d="M.816.483h135.377v35H.816z"
            fill="#fff"
          />
        </clippath>
      </defs>
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
